code/modules/mining/lavaland/loot/ashdragon_loot.dm

```diff
@@ -191,6 +191,14 @@
 	if(is_type_in_typecache(target, banned_turfs))
 		return
 
+	if(!is_mining_level(user.z)) //Will only spawn a few sparks if not on mining z level
+
+		if(world.time > timer)
```

Comment:
This is already checked at line 188. Don't need this here. Just need to set the timer

Reply:
Solved in 72437017dce98d8019ddb00da6ee04feab2be82a